### PR TITLE
Kibrantn/servicebus/unify track2 into master final

### PR DESCRIPTION
**Pull request by KieranBrantnerMagee**
Given the absence of a non-preview track1 release, there is no need to keep a separate v1 and v2 branch. Move all active work into master.
The individual commits have been reviewed, so eyeball this only with a mind for the integration of the v1 and t2 branches, and as a "scream test" for the overall consolidation.
I would also prefer to take this as a merge commit, as opposed to a squash, given the nature of this operation.

(Note: This is the proper version of [this PR](https://github.com/Azure/azure-sdk-for-python/pull/9389), transitioned so that we can do this as a merge and preserve history, in this instance I've cleaned up the work necessary to get the prior unification in alignment with CI)

### CODE REVIEW

File: sdk/servicebus/azure-servicebus/tests/servicebus_preparer.py [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+# are ported to offline-compatible code.
+def AreLiveTestsEnabled():
+    # This is how the CI pipe denotes that it's a live test run.
+    if os.environ.get('AZURE_RUN_MODE', None) == 'Live':
+        return True
+    # This is how local test runs denote that they're live.
+    # Note: This isn't amazing, because it could be present but not populated,
+    # but it's isomorphic to the old skips where it'd only check if the env vars were present.
+    try:
+        from devtools_testutils import mgmt_settings_real
```
Comment (lmazuel) [inline sdk/servicebus/azure-servicebus/tests/servicebus_preparer.py]:
```python
from devtools_testutils import is_live
return is_live
```
that will be more accurate

Comment (lmazuel) [reply sdk/servicebus/azure-servicebus/tests/servicebus_preparer.py]: if your CI also defines AZURE_TEST_RUN_LIVE, you can even also completly skip this method totally and use the live_test_only that will do exactly your skip if for free.
https://github.com/Azure/azure-sdk-for-python/blob/01447005686fab41d243d6bb6834653e1ce1fe33/sdk/storage/azure-storage-blob/tests/test_large_block_blob.py#L79

File: sdk/servicebus/azure-servicebus/tests/async_tests/test_queues_async.py [hunk 2]
```diff
@@ -65,990 +67,1164 @@ def print_message(message):
...
-        shared_access_key_name=live_servicebus_config['key_name'],
-        shared_access_key_value=live_servicebus_config['access_key'],
-        debug=False)
-
-    queue_client = client.get_queue(standard_queue)
-    async with queue_client.get_receiver(idle_timeout=5, mode=ReceiveSettleMode.PeekLock) as receiver:
+    @pytest.mark.liveTest
+    @pytest.mark.skipif(not AreLiveTestsEnabled(), reason="This test only runs against live resources")
+    @pytest.mark.asyncio
+    @ResourceGroupPreparer(name_prefix='servicebustest')
```
Comment (lmazuel) [inline sdk/servicebus/azure-servicebus/tests/async_tests/test_queues_async.py]: Please pass `random_name_enabled=True` to all your ResourceGroupPreparer, that's the value I want to make default once I have re-recorded all Mgmt tests correctly. New PRs should always pass this flag, and new preparer implementation should force it and not even expose it

File: sdk/servicebus/azure-servicebus/tests/servicebus_preparer.py [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+                 name_prefix='',
+                 sku='Standard', location='westus',
+                 parameter_name=SERVICEBUS_NAMESPACE_PARAM,
+                 resource_group_parameter_name=RESOURCE_GROUP_PARAM,
+                 disable_recording=True, playback_fake_resource=None,
+                 client_kwargs=None):
+        super(ServiceBusNamespacePreparer, self).__init__(name_prefix, 24,
+                                                          disable_recording=disable_recording,
+                                                          playback_fake_resource=playback_fake_resource,
+                                                          client_kwargs=client_kwargs)
```
Comment (lmazuel) [inline sdk/servicebus/azure-servicebus/tests/servicebus_preparer.py]: Please force `random_name_enabled=True` and don't even expose it.

File: sdk/servicebus/azure-servicebus/tests/async_tests/test_queues_async.py (hunk 2 above)
Comment (KieranBrantnerMagee) [reply sdk/servicebus/azure-servicebus/tests/async_tests/test_queues_async.py]: I put the partial within devtools_testutils to keep that logic centralized and make swapping out trivial.  Don't hesitate to shout if this is problematic.

**Pull request merged**